lib/dal/src/attribute/prototype/debug.rs

```diff
@@ -0,0 +1,207 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use thiserror::Error;
+
+use crate::attribute::prototype::{
+    argument::{
+        static_value::StaticArgumentValue,
+        value_source::{ValueSource, ValueSourceError},
+        AttributePrototypeArgument, AttributePrototypeArgumentError,
+    },
+    AttributePrototypeError,
+};
+use crate::attribute::value::{AttributeValueError, ValueIsFor};
+
+use crate::prop::{PropError, PropPath};
+
+use crate::func::execution::{FuncExecution, FuncExecutionError};
+use crate::socket::input::InputSocketError;
+use crate::socket::output::OutputSocketError;
+use crate::workspace_snapshot::node_weight::NodeWeightError;
+use crate::workspace_snapshot::WorkspaceSnapshotError;
+use crate::{
+    AttributePrototype, AttributePrototypeId, AttributeValue, AttributeValueId, Component,
+    ComponentError, DalContext, Func, FuncError, FuncId, InputSocket, Prop,
+};
+use serde_json::Value;
+use telemetry::prelude::*;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AttributePrototypeDebugView {
+    pub func_id: FuncId,
+    pub func_execution: Option<FuncExecution>,
+    pub prototype_id: AttributePrototypeId,
+    pub func_name: String,
+    pub func_args: HashMap<String, Vec<serde_json::Value>>,
+    pub arg_sources: HashMap<String, Option<String>>,
+    pub all_attribute_values: Vec<AttributeValueId>,
+}
+
+type AttributePrototypeDebugViewResult<T> = Result<T, AttributePrototypeDebugViewError>;
+
+#[remain::sorted]
+#[derive(Error, Debug)]
+pub enum AttributePrototypeDebugViewError {
+    #[error(transparent)]
+    AttributePrototypeArgumentError(#[from] AttributePrototypeArgumentError),
+    #[error(transparent)]
+    AttributePrototypeError(#[from] AttributePrototypeError),
+    #[error(transparent)]
+    AttributeValue(#[from] AttributeValueError),
+    #[error(transparent)]
+    ComponentError(#[from] ComponentError),
+    #[error(transparent)]
+    Func(#[from] FuncError),
+    #[error(transparent)]
+    FuncExecution(#[from] FuncExecutionError),
+    #[error(transparent)]
+    InputSocketError(#[from] InputSocketError),
+    #[error(transparent)]
+    NodeWeightError(#[from] NodeWeightError),
+    #[error(transparent)]
+    OutputSocketError(#[from] OutputSocketError),
+    #[error(transparent)]
+    PropError(#[from] PropError),
+    #[error(transparent)]
+    ValueSourceError(#[from] ValueSourceError),
+    #[error(transparent)]
+    WorkspaceSnapshotError(#[from] WorkspaceSnapshotError),
+}
+impl AttributePrototypeDebugView {
+    #[instrument(level = "debug", skip_all)]
+    pub async fn new_from_attribute_value(
```

Comment:
nit: The word `new` may imply some persistence. That's why we will often use `assemble` or `list`, depending on the object. That being said, Rust convention is that methods with the name `new` often return `Self`. I think if you renamed this to `new` and kept the parameters the same, that would work too.

Reply:
Funny enough, I had it as `assemble` originally!  Following my other comment, we'll also need to generate this for the socket debug views, so I had envisioned a `new_from_socket` func.... but maybe I just go back to `assemble` after-all and use the `AttributeValue` for that Socket to drive it 🤷‍♀️

Reply:
Whatever works best :smile:

I think either `new` or `assemble` or `assemble_<other-words>` works!